api/lib/infrastructure/mail-check.js

```diff
@@ -1,10 +1,10 @@
-const { promisify } = require('util');
-const dns = require('dns');
-const resolveMx = promisify(dns.resolveMx);
+const { Resolver } = require('node:dns').promises;
+
+const resolver = new Resolver();
 
 module.exports = {
   checkDomainIsValid(address) {
     const domain = address.replace(/.*@/g, '');
-    return resolveMx(domain).then(() => true);
+    return resolver.resolveMx(domain).then(() => true);
```

Comment:
En profiter pour être plus explicite sur le nom de l'argument de cette fonction, `address` n'est pas le nom d'argument auquel on s'attend le plus :

```suggestion
  checkDomainIsValid(emailAddress) {
    const domain = emailAddress.replace(/.*@/g, '');
    return resolver.resolveMx(domain).then(() => true);
```